NuRadioMC/utilities/inelasticities.py

```diff
@@ -23,58 +37,112 @@ def get_neutrino_inelasticity(n_events, rnd=None):
+                                          fill_value="extrapolate", kind="cubic")
+
+                    yyy = np.linspace(0, 1, 1000)
+                    yyy = 0.5 * (yyy[:-1] + yyy[1:])
+                    dsigma_dyy = 10 ** get_y(np.log10(yyy))
+
+                    yy[mask] = rnd.choice(yyy, size=size, p=dsigma_dyy / np.sum(dsigma_dyy))
+        return yy
 
-    return inelasticities
+    else:
+        raise AttributeError(f"inelasticity model {model} is not implemented.")
 
 
-def get_ccnc(n_events, rnd=None):
+def get_ccnc(n_events, rnd=None, model="hedis_bgr18", energy=None):
     """
     Get the nature of the interaction current: cc or nc
-    Ported from Shelf MC
 
     Parameters
     ----------
     n_events: int
         Number of events to be returned
-    rnd: random generator object
-        if None is provided, a new default random generator object is initialized
-        
+    rnd: random generator object (default: None)
+        If None is provided, a new default random generator object is initialized
+    model: string (default: "hedis_bgr18")
+        The cross section model to determine cc fraction. For options see cross_sections.py
+    energy: float or array (default: None)
+        Energy of the neutrino. If None is provided a constant value is used (only for CTW model).
+
     Returns
     -------
     ccnc: array
         Array with 'cc' or 'nc'
+
+    See Also
+    --------
+    NuRadioMC.utilities.cross_sections.get_nu_cross_section
     """
-    if(rnd is None):
-        rnd = np.random.default_rng()
+    rnd = rnd or np.random.default_rng()
+
     random_sequence = rnd.uniform(0., 1., n_events)
-    ccnc = []
-    for i, r in enumerate(random_sequence):
-        #    if (r <= 0.6865254):#from AraSim
-        if(r <= 0.7064):
-            ccnc.append('cc')
-        else:
-            ccnc.append('nc')
 
-    return np.array(ccnc)
+    if energy is None:
+        assert model.lower() == "ctw", "Only CTW supports energy-independent cc/nc fraction Energy is required for BGR18 model"
+        # Ported from Shelf MC
+        # https://github.com/persic/ShelfMC/blob/daf56916d85de019e848f415c2e9f4643a744674/functions.cc#L1055-L1064
+        # based on CTW cross sections https://link.aps.org/doi/10.1103/PhysRevD.83.113009
+        ccnc = np.where(random_sequence <= 0.7064, 'cc', 'nc')
+    else:
+        if not isinstance(energy, (float, int)):
+            assert len(energy) == n_events, "Energy must be a scalar or an array of the same length as n_events"
+
+        # Flavor only relevant to determine if its a neutrino or antineutrino. For cross section ratio it
+        # hopefully doesn't matter
+        cc = cross_sections.get_nu_cross_section(energy, flavors=12, inttype="cc", cross_section_type=model.lower())
+        nc = cross_sections.get_nu_cross_section(energy, flavors=12, inttype="nc", cross_section_type=model.lower())
```

Comment:
It's a smallish effect... but it does seem to matter? At least neutrino vs antineutrino. See the plot here: https://github.com/nu-radio/NuRadioMC/issues/600#issue-2026805640. I would add it as an optional parameter to the function.